src/openrct2-ui/windows/Changelog.cpp

```diff
@@ -41,273 +41,308 @@ static constexpr const rct_string_id WINDOW_TITLE = STR_CHANGELOG_TITLE;
+            ProcessChangelogText(_newVersionInfo->changelog);
+        }
+        else
+        {
+            log_error("ChangelogWindow: Could not process NewVersionInfo, result was undefined");
+        }
     }
-}
-
-static void window_new_version_process_info()
-{
-    _newVersionInfo = GetContext()->GetNewVersionInfo();
-
-    char version_info[256];
-
-    const char* version_info_ptr = _newVersionInfo->name.c_str();
-    format_string(version_info, 256, STR_NEW_RELEASE_VERSION_INFO, &version_info_ptr);
-
-    _changelogLines.emplace_back(version_info);
-    _changelogLines.emplace_back("");
-
-    window_changelog_process_changelog_text(_newVersionInfo->changelog);
-}
-
-static void window_changelog_dispose_data()
-{
-    _changelogLines.clear();
-    _changelogLines.shrink_to_fit();
-}
 
-static std::string GetChangelogPath()
-{
-    auto env = GetContext()->GetPlatformEnvironment();
-    return env->GetFilePath(PATHID::CHANGELOG);
-}
-
-static std::string GetChangelogText()
-{
-    auto path = GetChangelogPath();
-#if defined(_WIN32) && !defined(__MINGW32__)
-    auto pathW = String::ToWideChar(path);
-    auto fs = std::ifstream(pathW, std::ios::in);
-#else
-    auto fs = std::ifstream(path, std::ios::in);
-#endif
-    if (!fs.is_open())
+    /**
+     * @brief Clears out stored changelog line-data and shrinks store capacity
+     *
+     */
+    void DisposeChangelogData()
     {
-        throw std::runtime_error("Unable to open " + path);
+        _changelogLines.clear();
+        _changelogLines.shrink_to_fit();
     }
-    return std::string((std::istreambuf_iterator<char>(fs)), std::istreambuf_iterator<char>());
-}
 
-static bool window_changelog_read_file()
-{
-    std::string _changelogText;
-    try
+    /**
+     * @brief Get the absolute path for the changelog file as a std::string
+     *
+     * @return std::string
+     */
+    std::string GetChangelogPath()
     {
-        _changelogText = GetChangelogText();
+        auto env = GetContext()->GetPlatformEnvironment();
+        return env->GetFilePath(PATHID::CHANGELOG);
     }
-    catch (const std::bad_alloc&)
+
+    /**
+     * @brief Attempts to read the changelog file, returns true on success
+     *
+     * @return true
+     * @return false
+     */
+    bool ReadChangelogFile()
     {
-        log_error("Unable to allocate memory for changelog.txt");
-        return false;
+        std::string _changelogText;
+        try
+        {
+            _changelogText = GetChangelogText();
+        }
+        catch (const std::bad_alloc&)
+        {
+            log_error("Unable to allocate memory for changelog.txt");
+            return false;
+        }
+        catch (const std::exception&)
+        {
+            log_error("Unable to read changelog.txt");
+            return false;
+        }
+
+        ProcessChangelogText(_changelogText);
+        return true;
     }
-    catch (const std::exception&)
+
+    /**
+     * @brief Pushes a line of text to the changelog
```

Comment:
Does not reflect what the function actually does.

Reply:
I've updated this brief, please let me know if it's too verbose and I can cut it down a bit more.